lib/main.js

```diff
@@ -15,104 +15,133 @@ let Matrix = require('./models/matrix');
-        this.Interval = Interval;
+    this.tune = new Tune();
+    this.note = this.tune.note.bind(this.tune);
 
-        this.colors = {
-          accent: '#2bb',
-          fill: '#eee',
-          light: '#fff',
-          dark: '#333',
-          mediumLight: '#ccc',
-          mediumDark: '#666'
-        };
+    this._clock = null;
+    this.Interval = Interval;
 
-        this.transform = Transform;
-        this.add = Transform.add;
+    this.colors = {
+      accent: '#2bb',
+      fill: '#eee',
+      light: '#fff',
+      dark: '#333',
+      mediumLight: '#ccc',
+      mediumDark: '#666'
+    };
 
+    this.transform = Transform;
+    this.add = Transform.add;
 
-        this.Add = {};
-        for (let key in Interfaces) {
-          this.Add[key] = Transform.add.bind(this,key);
-        }
 
+    this.Add = {};
+    for (let key in Interfaces) {
+      this.Add[key] = Transform.add.bind(this, key);
+    }
 
 
 
-        /* create default component size */
-        /* jshint ignore:start */
-        var existingStylesheets = document.getElementsByTagName("style");
-        var defaultSizeDeclaration = '[nexus-ui]{height:5000px;width:5000px}';
-        var defaultStyleNode = document.createElement('style');
-        defaultStyleNode.type = 'text/css';
-        defaultStyleNode.innerHTML = defaultSizeDeclaration;
-        if (existingStylesheets.length > 0) {
-          var parent = existingStylesheets[0].parentNode
-          parent.insertBefore( defaultStyleNode, existingStylesheets[0])
-        } else {
-          document.write('<style>'+defaultSizeDeclaration+'<\/style>');
-        }
-        /* jshint ignore:end */
 
+    /* create default component size */
+    /* jshint ignore:start */
+    var existingStylesheets = document.getElementsByTagName("style");
+    var defaultSizeDeclaration = '[nexus-ui]{height:5000px;width:5000px}';
+    var defaultStyleNode = document.createElement('style');
+    defaultStyleNode.type = 'text/css';
+    defaultStyleNode.innerHTML = defaultSizeDeclaration;
+    if (existingStylesheets.length > 0) {
+      var parent = existingStylesheets[0].parentNode
+      parent.insertBefore(defaultStyleNode, existingStylesheets[0])
+    } else {
+      document.write('<style>' + defaultSizeDeclaration + '<\/style>');
     }
+    /* jshint ignore:end */
 
-    get context() {
-      return this._context;
-    }
+  }
 
-    set context(ctx) {
-      this.clock.stop();
-      this._context = ctx;
-      this.clock = new WAAClock(this.context);
-      this.clock.start();
+  get context() {
+    if (!this._context) {
+      this._context = new DefaultContext();
     }
+    return this._context;
+  }
 
+  set context(ctx) {
+    if (this._clock) {
+      this._clock.stop();
+      this._clock = null;
+    }
 
+    this._context = ctx;
+    unlockAudioContext(ctx);
+  }
 
+  get clock() {
+    if (!this._clock) {
+      this._clock = new WAAClock(this.context);
+      this._clock.start();
```

Comment:
Clock is only created (and started) when it is specifically requested. I didn't create a setter.